sklearn/metrics/pairwise.py

```diff
@@ -1049,6 +1049,11 @@ def distance_metrics():
     return PAIRWISE_DISTANCE_FUNCTIONS
 
 
+def _dist_wrapper(dist_func, dist_matrix, slice, *args, **kwargs):
+    """Write in-place to a slice of a distance matrix"""
+    dist_matrix[:, slice] = dist_func(*args, **kwargs)
```

Comment:
Ok, then maybe call it `slice_` to avoid name collision with a reserved key word?